### PR TITLE
Alternative Event Tracking: `withTrackedReducer`

### PR DESCRIPTION
**Pull request by rainerhahnekamp**
## Summary

This PR introduces `withTrackedReducer` as an alternative approach to tracking reducer-based state changes in Redux DevTools. It addresses limitations in the original `withEventsTracking` implementation.

The original implementation (based on #231) attempted to automatically track reducer-based state changes via the glitch tracker. However, this approach has fundamental limitations that cannot be resolved with the specified approach.

Several edge cases are documented in the skipped tests within `with-events-tracking.spec.ts`.

This PR introduces an alternative way, where end users have to use the wrapperd function `withTrackedReducer` which calls internally `updateState` using the event's name.

If we proceed with this approach, the following items need to be completed:

- [ ] Update documentation to explain `withTrackedReducer` usage and when to use it vs `withReducer`
- [ ] Update demo applications to showcase the new API
- [ ] Remove or deprecate code from the initial `withEventsTracking` approach (if applicable)
- [ ] Add validation/runtime checks to ensure `withTrackedReducer` is only used with stores that have devtools and glitch tracking enabled

### CODE REVIEW

File: apps/demo/src/app/events-sample/events-sample.component.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+          <input
+            matInput
+            [(ngModel)]="filterText"
+            (ngModelChange)="dispatch.filterUpdated({ filter: $event })"
+            placeholder="Filter by title or author..."
+          />
+          <mat-icon matSuffix>search</mat-icon>
+        </mat-form-field>
+
+        <button mat-raised-button color="primary" (click)="addRandomBook()">
```
Comment (michael-small) [inline apps/demo/src/app/events-sample/events-sample.component.ts]: nit: though the demo is lowkey, I would like things like requiring a `type` on `button` elements like this. I am going about backfilling this for existing demo code.

File: apps/demo/src/app/events-sample/events-sample.component.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { injectDispatch } from '@ngrx/signals/events';
+import { bookEvents } from './book-events';
+import { BookStore } from './book.store';
+
+@Component({
+  selector: 'demo-events-sample',
+  standalone: true,
```
Comment (michael-small) [inline apps/demo/src/app/events-sample/events-sample.component.ts]:
```suggestion
```

**Comment by rainerhahnekamp** [comment]
@michael-small @wolfmanfx 

I see it as temporarily. In the end, we need support for meta-reducers in the core...

File: libs/ngrx-toolkit/src/lib/devtools/with-tracked-reducer.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  SignalStoreFeature,
+  signalStoreFeature,
+  type,
+} from '@ngrx/signals';
+import { ReducerEvents, withEventHandlers } from '@ngrx/signals/events';
+import { tap } from 'rxjs/operators';
+import { updateState } from './update-state';
+
+export function withTrackedReducer<State extends object>(
+  ...caseReducers: CaseReducerResult<State, any>[]
```
Comment (rainerhahnekamp) [inline libs/ngrx-toolkit/src/lib/devtools/with-tracked-reducer.ts]: For some reason, I had to use `any` here. The `EventCreator` - as it is in the original version - didn't work. I guess that this is due to different settings in the `tsconfig.json`. 

Unless, we won't get meta reducers, I wouldn't investigate this any further. The only thing I would change is to replace `any` with `unknown`.

**Review by wolfmanfx — APPROVED** [review]
